SimpleAkkaDotNetCluster/SimpleClusterLib/EchoActor.cs

```diff
@@ -6,14 +6,22 @@ public class EchoActor : UntypedActor
     {
         public EchoActor()
         {
-            
             DisplayHelper.WriteLine("Echo actor created @ " + this.Self.Path.Address);
         }
 
         protected override void OnReceive(object message)
         {
-            DisplayHelper.WriteLine("EchoActor Message in " + message);
-            Sender.Tell("Server returns:" + message);
+            switch (message)
+            {
+                case string str:
+                    DisplayHelper.WriteLine("EchoActor Message in " + str);
+                    Sender.Tell("Server returns:" + str);
+                    break;
+                case PayloadEnvelope envelope:
```

Comment:
Note the additional code here. We're allowing the `EchoActor` to receive a `PayloadEnvelope`. This will be forwarded by the `ClusterClientReceptionist` to the actor, and we're using the actor reference inside of it as the sender of the message.